src/vos/vos_dtx.c

```diff
@@ -195,14 +195,14 @@ static int
 dtx_act_ent_update(struct btr_instance *tins, struct btr_record *rec,
 		   d_iov_t *key, d_iov_t *val)
 {
-	/* It is possible that when commit the DTX for the first time,
-	 * it failed at removing the DTX entry from active table, but
-	 * at that time the DTX entry has already been added into the
-	 * committed table that is in DRAM. Currently, we do not have
-	 * efficient way to recover such DRAM based btree structure,
-	 * so just keep it there. Then when we re-commit such DTX, we
-	 * may come here.
-	 */
+	struct vos_dtx_act_ent	*dae_new = val->iov_buf;
+	struct vos_dtx_act_ent	*dae_old;
+
+	dae_old = umem_off2ptr(&tins->ti_umm, rec->rec_off);
+	D_ASSERTF(0, "NOT allow to update act DTX entry for "DF_DTI
+		  " from epoch %lu to %lu\n", DP_DTI(&DAE_XID(dae_old)),
```

Comment:
I've been changing epochs to DF_X64 since you used that in a lot of place in dtx...we should be consisitent.   I think hex is superior because it's an HLC so we can see more easily logical differences.

Reply:
fixed.